src/box/session.h

```diff
@@ -87,6 +88,14 @@ struct session_meta {
 		/** Console file/socket descriptor. */
 		int fd;
 	};
+	struct {
+		/** Peer address or NULL if the session is local. */
+		struct sockaddr *addr;
```

Comment:
Could also use `addrlen == 0` as a sign whether the session is local. Then this member wouldn't be needed.

Reply:
Yes, this looks better. Thanks for the suggestion.

<details><summary>diff</summary>

```diff
diff --git a/src/box/session.cc b/src/box/session.cc
index 801e8c2389cc..5cd0cf88f3a6 100644
--- a/src/box/session.cc
+++ b/src/box/session.cc
@@ -392,9 +392,9 @@ session_find(uint64_t sid)
 const char *
 session_peer(struct session *session)
 {
-	if (session->meta.peer.addr == NULL)
+	if (session->meta.peer.addrlen == 0)
 		return NULL;
-	return sio_strfaddr(session->meta.peer.addr,
+	return sio_strfaddr(&session->meta.peer.addr,
 			    session->meta.peer.addrlen);
 }
 
diff --git a/src/box/session.h b/src/box/session.h
index 6e8dde85d1ff..2faa8aa43f04 100644
--- a/src/box/session.h
+++ b/src/box/session.h
@@ -89,12 +89,14 @@ struct session_meta {
 		int fd;
 	};
 	struct {
-		/** Peer address or NULL if the session is local. */
-		struct sockaddr *addr;
-		/** Peer address size. */
+		union {
+			/** Peer address. */
+			struct sockaddr addr;
+			/** Peer address storage. */
+			struct sockaddr_storage addrstorage;
+		};
+		/** Peer address size or 0 if the session is local. */
 		socklen_t addrlen;
-		/** Peer address storage. */
-		struct sockaddr_storage addrstorage;
 	} peer;
 	/** Console output format. */
 	enum output_format output_format;
@@ -349,8 +351,6 @@ session_set_peer_addr(struct session *session,
 {
 	assert(addrlen <= sizeof(session->meta.peer.addrstorage));
 	memcpy(&session->meta.peer.addrstorage, addr, addrlen);
-	session->meta.peer.addr =
-		(struct sockaddr *)&session->meta.peer.addrstorage;
 	session->meta.peer.addrlen = addrlen;
 }
```

</details>